### PR TITLE
prefix metrics with restate_

### PR DESCRIPTION
**Pull request by AhmedSoliman**
prefix metrics with restate_

We should namespace all our metrics with the application name, additionally, grouping invoker task counters and using status as label instead of a counter per status (better ergonomics)

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/restatedev/restate/pull/1077).
* __->__ #1077
* #1073

### CODE REVIEW

File: crates/invoker-impl/src/metric_definitions.rs [hunk 1]
```diff
@@ -26,26 +27,8 @@ pub(crate) fn describe_metrics() {
...
-    );
-
-    describe_counter!(
-        INVOKER_INVOCATION_TASK_SUSPENDED,
-        Unit::Count,
-        "Number of active invocation tasks suspended"
-    );
-
-    describe_gauge!(
-        INVOKER_INFLIGHT_INVOCATIONS,
```
Comment (tillrohrmann) [inline crates/invoker-impl/src/metric_definitions.rs:47]: W/o this gauge we no longer know how many invocations are currently executed by the invoker, right? Isn't this a useful metric to have or why did you remove this metric?

**Review by tillrohrmann — COMMENTED** [review]
The changes makes sense to me. I had a question concerning the removal of the `INVOKER_INFLIGHT_INVOCATIONS` metric which seems somewhat useful.

File: crates/invoker-impl/src/metric_definitions.rs (hunk 1 above)
Comment (AhmedSoliman) [reply crates/invoker-impl/src/metric_definitions.rs:47]: Yes, it's a useful metric but unfortunately it's broken (shows mostly negative results in my tests) with the current implementation. We should re-introduce it again but it needs to cover the correct spots to increment and decrement.

Comment (tillrohrmann) [reply crates/invoker-impl/src/metric_definitions.rs:47]: Thx for the clarification.

**Review by tillrohrmann — APPROVED** [review]
LGTM. +1 for merging.
